File: llm_generated_code/Gemini/CPP/Level1/Q11.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

struct Cell {
  int row;
  int col;
};

int trappedLandCells(const vector<vector<int>>& grid) {
  int m = grid.size();
  int n = grid[0].size();

  int trappedCells = 0;
  vector<vector<bool>> visited(m, vector<bool>(n, false));

  // Perform DFS starting from land cells on the boundary
  for (int i = 0; i < m; ++i) {
    for (int j = 0; j < n; ++j) {
      if (grid[i][j] == 1 && (i == 0 || j == 0 || i == m - 1 || j == n - 1)) {
        dfs(grid, visited, i, j);
      }
    }
  }

  // Count unvisited land cells
  for (int i = 0; i < m; ++i) {
    for (int j = 0; j < n; ++j) {
      if (grid[i][j] == 1 && !visited[i][j]) {
        trappedCells++;
      }
    }
  }

  return trappedCells;
}

void dfs(const vector<vector<int>>& grid, vector<vector<bool>>& visited, int row, int col) {
  visited[row][col] = true;

  int directions[4][2] = {{0, 1}, {0, -1}, {1, 0}, {-1, 0}};

  for (int i = 0; i < 4; ++i) {
    int newRow = row + directions[i][0];
    int newCol = col + directions[i][1];

    if (newRow >= 0 && newRow < grid.size() && newCol >= 0 && newCol < grid[0].size() &&
        grid[newRow][newCol] == 1 && !visited[newRow][newCol]) {
      dfs(grid, visited, newRow, newCol);
    }
  }
}

int main() {
  vector<vector<int>> grid = {
    {0, 1, 0, 0},
    {1, 0, 0, 1},
    {0, 1, 1, 0},
    {0, 0, 0, 0}
  };

  int trappedCells = trappedLandCells(grid);
  cout << "Number of trapped land cells: " << trappedCells << endl;

  return 0;
}

File: llm_generated_code/Gemini/Python/Level1/Q11.py
def num_enclaves(grid):
  """
  Finds the number of land cells where you cannot walk off the boundary.

  Args:
      grid: A list of lists representing the binary matrix.

  Returns:
      The number of land cells (enclaves)
  """

  rows, cols = len(grid), len(grid[0])
  directions = [(0, 1), (0, -1), (1, 0), (-1, 0)]  # Up, down, right, left

  def dfs(row, col):
    # Depth-First Search to mark connected land cells as visited

    if 0 <= row < rows and 0 <= col < cols and grid[row][col] == 1:
      grid[row][col] = 0  # Mark visited
      for dr, dc in directions:
        dfs(row + dr, col + dc)

  # Check boundary cells for land
  for i in range(rows):
    dfs(i, 0)  # Check first column
    dfs(i, cols - 1)  # Check last column

  for j in range(cols):
    dfs(0, j)  # Check first row
    dfs(rows - 1, j)  # Check last row

  # Count remaining land cells (enclaves)
  enclave_count = 0
  for i in range(rows):
    for j in range(cols):
      if grid[i][j] == 1:
        enclave_count += 1

  return enclave_count

# Example usage
grid = [[0, 0, 0, 0], [1, 0, 1, 0], [0, 1, 1, 0], [0, 0, 0, 0]]
enclaves = num_enclaves(grid)
print("Number of enclaves:", enclaves)
